### ISSUE TITLE
[BUG] Clicking 3 dot menu for connection just closes menu instead of taking the action

### ISSUE THREAD

**Issue by digitaldrummerj** [bitfocus/companion#2965]
### Is this a bug in companion itself or a module?

- [X] I believe this to be a bug in companion

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

On Windows 11, on both Edge and Chrome, when I click on the three-dot menu for a connection, it opens the menu, but when I click on any of the actions, it does nothing other than close the menu.  

### Steps To Reproduce

1. Add a connection
2. Click on the 3 dots menu for the connection
3. try to pick any of the options and watch it do nothing instead of opening help, opening variables or opening logs or deleting connection

### Expected Behavior

each of the menu items do the action they are suppose to do

### Environment (please complete the following information)

```markdown
- OS: Windows 11
- Browser: Chrome 126.0.6478.185, Edge 127.0.2651.74, Firefox 128.0.3
- Companion Version: v3.4.0+7161-main-e0c4cbdd
```

**Comment by peternewman** [comment]
Confirmed I've seen this too on a beta build but didn't get a chance to raise an issue.

**Comment by Julusian** [comment]
Huh, I am only able to reproduce this when using tap-to-click on a touchpad/trackpad but not the click buttons of a mouse of the same touchpad/trackpad.

Is anyone else able to confirm this?

**Comment by digitaldrummerj** [comment]
@Julusian I updated to beta v3.4.0-7177 and the issue is still happening.  Windows 11 and Chrome 127.0.6533.90.  I attached a quick little video showing the issue.

![2024-08-07_07-22-06](https://github.com/user-attachments/assets/d48d8af1-5d10-4737-b08a-8d2971874151)

**Comment by peternewman** [comment]
> Huh, I am only able to reproduce this when using tap-to-click on a touchpad/trackpad but not the click buttons of a mouse of the same touchpad/trackpad.

On:
Edition	Windows 10 Enterprise
Version	22H2
OS build	19045.4651
Experience	Windows Feature Experience Pack 1000.19060.1000.0

On both Chrome 127.0.6533.89 and Edge 127.0.2651.86 (Official build) (64-bit) with a mouse plugged into a desktop I agree I don't see it.

**Comment by peternewman** [comment]
Pointing at the same Companion install from a laptop with a trackpad, I sometimes have issues (but it does seem to sometimes work, perhaps like you after I've used the trackpad buttons). But using the same mouse into the laptop it works fine.

**Comment by digitaldrummerj** [comment]
Here are some additional testing steps that I did:

1. I remotely accessed my Mac Mini M2 using Jump Desktop.
2. I accessed the Companion GUI using both Chrome and Safari.
3. I attempted to open the logs or known issues on the first several connections. Nothing happened.
4. After several attempts, it suddenly started working.
5. Once it started working, it was somewhat consistent, but it still didn't work sometimes when I clicked on an option.
6. I also noticed that in Safari, the options opened in another window instead of another tab.

I tried using two different mice (Razer Basilisk v3 wired and Corsair M65 Elite), but both had the same issue. I'm using a desktop machine, so there's no trackpad involved.

I also tried clearing my cache for the Companion GUI and reset my whole configuration. I still had the same issue. This Companion instance has been around since the v3 betas and has been upgraded to each stable release along the way.

I like the new menu, but it seems a bit finicky and not consistently working.

anything else for troubleshooting that you would like me to do?

**Comment by Julusian** [comment]
I think I have a fix for this.

It appears to have been caused by the popover closing itself when losing focus, even when focus is lost to a child element.
This results in the mouseup event sometimes hitting whatever was behind the popover, rather than the button you intend to click.
So if you click fast it would work, a slow click wouldnt. And I saw this with tap clicking because that uses I assume a fixed duration between the down and up which looked (on my machine) to be too slow, causing the up to be lost.

The fix feels bad but is simply to react to the mousedown event, as we always receive that

**Comment by digitaldrummerj** [comment]
@Julusian the latest beta is better.  If I tap the mouse button it works every time but if I do a full press of the mouse button or hold the mouse button down too long, then the help popup instantly closes when the window is not big enough to be under my mouse cursor and the delete popup always instantly closes as it is never under my mouse button.

**Comment by Julusian** [comment]
I've tried a couple of windows 11 machines, and I'm not able to reproduce any remaining issues

**Comment by phillipivan** [comment]
I believe I am encountering this issue, or a variation thereof when trying to delete connections. 
When I click the 3 Dot menu, and select delete, the delete confirmation window momentarily flashes up then is gone.
Is I hold the mouse button down, when selecting the delete option, the delete window stays preset until I release the left mouse button. If I move the cursor over the Delete confirmation button I am able to activate it and delete the module. Subsequent attempts to delete connections then seem fine until Companion is restarted. I have not noticed this behavior from other menu items such as Logs or Help.

System Info:
- Windows 10
- Chrome
- Companion 3.4 Stable

**Comment by Coupe70** [comment]
I'll move this over from my previous post and add some things:
v3.4.1 + 7223
I was able to get persistent "are you sure?" dialogs for the 2 upper connections in the list.
This seems to work for a small area at the top, no matter which connections I move there by scrolling (when having many connections) and seems to confirm what you said about the focus: For the 2 upper connections the menu item 'delete' is in the screen area where the dialog window will appear, so it appears under my mouse pointer and gets hit by the mouse-up event (in my screen resolution).
![grafik](https://github.com/user-attachments/assets/e36f805a-bb06-4bc9-a462-1ca5b78ebe98)
I can also confirm that a fast click on 'delete' meu item works for connections in other screen positions, but it has to be VERY fast.

**Comment by Julusian** [comment]
This is fixed in the v3.4.2 release

**Comment by Coupe70** [comment]
I can confirm that for me the mentioned behaviour does not occur anymore in 3.4.2 7335.
When I click somewhere outside of the dialog box it still disappears though.
Doesn't do any harm, just wanted to mention it as it's a bit uncommon behaviour.
